twake/frontend/src/app/components/ChatUploads/PendingFileComponents/PendingFileRow.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { Row, Col, Typography, Divider, Progress, Button, Tooltip } from 'antd';
+import { Pause, Play, X } from 'react-feather';
+import { PendingFileStateType, PendingFileType } from 'app/models/File';
+import ChatUploadService from '../ChatUploadService';
+import {
+  isPendingFileStatusError,
+  isPendingFileStatusPause,
+  isPendingFileStatusSuccess,
+} from '../utils/PendingFiles';
+import Languages from 'services/languages/languages';
+
+type PropsType = {
+  pendingFileState: PendingFileStateType;
+  pendingFile: PendingFileType;
+};
+
+const { Text } = Typography;
+export default ({ pendingFileState, pendingFile }: PropsType) => {
+  const getProgressStrokeColor = (status: PendingFileStateType['status']) => {
+    if (isPendingFileStatusError(status)) return 'var(--error)';
+    if (isPendingFileStatusPause(status)) return 'var(--warning)';
+
+    return 'var(--success)';
+  };
+
+  const pauseOrResume = (id: PendingFileStateType['id']) => ChatUploadService.pauseOrResume(id);
+
+  const cancel = (id: PendingFileStateType['id']) => ChatUploadService.cancel(id);
+
```

Comment:
The idea of the useChatUpload hook is to be used also here. I think that the hook API is not the right one or that we need more than one. We should be able to do something like

```js
const uploadState = useChatUploadState(id); // id is the upload id

// and be able to do this

uploadState.pauseOrResume();
uploadState.cancel();
```

Reply:
Done